types/scratch-vm/index.d.ts

```diff
@@ -0,0 +1,199 @@
+    func?: string;
+
+    /** The type of block (command, reporter, etc.) being described. */
+    blockType: ValueOf<Scratch.extensions.BlockType>;
+
+    /** The text on the block, with [PLACEHOLDERS] for arguments. */
+    text: string;
+
+    /** Whether this block should not appear in the block palette. */
+    hideFromPalette?: boolean;
+
+    /** Whether the block ends a stack - no blocks can be connected after it. */
+    isTerminal?: boolean;
+
+    /** Whether this block is a reporter but should not allow a monitor. */
+    disableMonitor?: boolean;
+
+    /** If this block is a reporter, this is the scope/context for its value. */
+    reporterScope?: ReporterScope;
+
+    /** Whether a hat block is edge-activated. */
+    isEdgeActivated?: boolean;
+
+    /** Whether a hat/event block should restart existing threads. */
+    shouldRestartExistingThreads?: boolean;
+
+    /** For flow control blocks, the number of branches/substacks for this block. */
+    branchCount?: number;
+
+    /** Map of argument placeholder to metadata about each arg. */
+    arguments?: Record<string, ExtensionArgumentMetadata>;
+}
+
+/** All the metadata needed to register an argument for an extension block. */
+interface ExtensionArgumentMetadata {
+    /** The type of the argument (number, string, etc.) */
+    type: ValueOf<Scratch.extensions.ArgumentType>;
+
+    /** The default value of this argument. */
+    defaultValue?: any;
+
+    /** The name of the menu to use for this argument, if any. */
+    menu?: string;
+}
+
+/** All the metadata needed to register an extension drop-down menu. */
+type ExtensionMenuMetadata = ExtensionDynamicMenu | ExtensionMenuItems;
+
+/** The string name of a function which returns menu items. */
+type ExtensionDynamicMenu = string;
+
+/** Items in an extension menu. */
+type ExtensionMenuItems = Array<ExtensionMenuItemSimple | ExtensionMenuItemComplex>;
+
+/** A menu item for which the label and value are identical strings. */
+type ExtensionMenuItemSimple = string;
+
+/** A menu item for which the label and value can differ. */
+interface ExtensionMenuItemComplex {
+    /** The value of the block argument when this menu item is selected. */
+    value: any;
+
+    /** The human-readable label of this menu item in the menu. */
+    text: string;
+}
+
+declare class ScratchExtension {
+    /** Returns data about the extension. */
+    getInfo(): ExtensionMetadata;
+}
+
+/** Indicate the scope for a reporter's value. */
+declare enum ReporterScope {
+    /** This reporter's value is global and does not depend on context. */
+    GLOBAL = "global",
+
+    /**
+     * This reporter's value is specific to a particular target/sprite.
+     * Another target may have a different value or may not even have a value.
+     */
+    TARGET = "target"
+}
+
+declare namespace Scratch {
+    namespace extensions {
+        /** Block argument types. */
+        enum ArgumentType {
```

Comment:
THis IMO should be directly on the Scratch object, as per implementation:
```js
global.Scratch = global.Scratch || {};
global.Scratch.ArgumentType = ArgumentType;
global.Scratch.BlockType = BlockType;
global.Scratch.TargetType = TargetType;
```
the extensions is a special namespace created just to offer proxied 'register' method alias:
https://github.com/LLK/scratch-vm/blob/0dffc65ce99307d048f6b9a10b1c31b01ab0133d/src/extension-support/extension-worker.js#L57

Reply:
Fixed!